server/src/frameworks/Hardhat/HardhatProject.ts

```diff
@@ -47,7 +47,7 @@ export enum WorkerStatus {
 const REQUEST_TIMEOUT = 5000;
 
 export class HardhatProject extends Project {
-  public priority = 2;
+  public priority = 3;
```

Comment:
Can you walk me through your current mental model of how priority works? Why is hardhat 3 for instance?

Reply:
The adapter with the highest priority will have claim over a source file in case they are competing. I'm updating hardhat to priority = 4 (to avoid clashing with truffle, as ape was being developed in parallel). This would end up in the following order of priority: Hardhat > Truffle > Foundry > Ape > Projectless.